### ISSUE TITLE
Move network related components from MdeModulePkg to NetworkPkg (Bugzilla Bug 1293)

### ISSUE THREAD

**Issue by tianocore-issues** [tianocore/edk2#7604]
This issue was created automatically with [bugzilla2github](https://github.com/makubacki/bugzilla2github/tree/bz2gh_tianocore)
# Bugzilla Bug 1293
Date: 2018-11-02T03:31:46+00:00
From: sfu5 &lt;<siyuan.fu>&gt;
To:   sfu5 &lt;<siyuan.fu>&gt;
CC:   @lersek, nobody, @leiflindholm

Blocker for:  #1009, #1357
Last updated: 2019-06-24T11:29:12+00:00

**Comment by tianocore-issues** [comment]
## Comment 5455
Date: 2018-11-02 03:31:46 +0000
From: sfu5 &lt;<siyuan.fu>&gt;

- Industry Specification: ---
- Target OS: ---
- Bugzilla Assignee(s): sfu5 &lt;<siyuan.fu>&gt;

Bug #1278 &quot;Delete IPv4 only TCP/iSCSI/PXE drivers in MdeModulePkg&quot; has been confirmed to be included in edk2-stable201903 tag, platform DSC/FDF need to be updated to adapt the change by that time. This might be an opportune moment to make some more changes - move all edk2 network related drivers/libraries from MdeModulePkg to NetworkPkg - which will make the code structure easier to read and maintain in the future.

**Comment by tianocore-issues** [comment]
## Comment 5546
Date: 2018-11-08 10:07:14 +0000
From: @leiflindholm

I think it would be useful to schedule this change for _after_ the introduction of some sort of &quot;configuration fragments&quot; (centralized .dsc/.fdf include files). Once that is in place, the migration could be performed atomically, without any changes required to platform ports (because that will have happened as part of the configuration fragment work).

**Comment by tianocore-issues** [comment]
## Comment 5596
Date: 2018-11-14 21:54:43 +0000
From: sfu5 &lt;<siyuan.fu>&gt;

(In reply to Leif Lindholm from comment #1)
&gt; I think it would be useful to schedule this change for _after_ the
&gt; introduction of some sort of &quot;configuration fragments&quot; (centralized
&gt; .dsc/.fdf include files). Once that is in place, the migration could be
&gt; performed atomically, without any changes required to platform ports
&gt; (because that will have happened as part of the configuration fragment work).

Good suggestion, the DSC/FDF segment file could also contain some predefined symbols to allow simple configuration, like only include IP4 drivers.
I will create a patch for this.

**Comment by tianocore-issues** [comment]
## Comment 5697
Date: 2018-11-23 06:47:33 +0000
From: @lersek

Series from Siyuan on the list:

[edk2] [PATCH 0/6] Add DSC/FDF include segment files for network stack
https://lists.01.org/pipermail/edk2-devel/2018-November/032759.html

[edk2] [PATCH v2 0/6] Add DSC/FDF include segment files for network stack
https://lists.01.org/pipermail/edk2-devel/2018-November/032824.html

**Comment by tianocore-issues** [comment]
## Comment 5699
Date: 2018-11-23 06:58:40 +0000
From: @lersek

Siyuan, please don&apos;t forget to reference this BZ under

https://github.com/tianocore/tianocore.github.io/wiki/EDK-II-Release-Planning#edk2-stable201903-tag-planning

in particular, I suggest mentioning that build scripts / build command lines must be updated to account for the new -D flags.

**Comment by tianocore-issues** [comment]
## Comment 6637
Date: 2019-03-07 00:41:21 +0000
From: sfu5 &lt;<siyuan.fu>&gt;

Created attachment 267
Move network modules to NetworkPkg

Attach the proposed patch of the code movement. This is a incompatible change to platform, and the patch doesn&apos;t include edk2 platform change.

Attachment: [BZ-267-patch.zip](https://github.com/tianocore/user-attachments/raw/refs/heads/main/tianocore/edk2/BZ-267-patch.zip)

**Comment by tianocore-issues** [comment]
## Comment 7530
Date: 2019-05-14 10:39:57 +0000
From: @lersek

Further related patch series:

[edk2-devel] [Patch v3 0/3] Add package level include DSC/FDF in NetworkPkg
http://mid.mail-archive.com/20190425123735.10960-1-liming.gao@intel.com
https://edk2.groups.io/g/devel/message/39553

[edk2-devel] [Patch v4 0/3] Add package level include DSC/FDF in NetworkPkg
http://mid.mail-archive.com/20190429145012.14852-1-liming.gao@intel.com
https://edk2.groups.io/g/devel/message/39779

[edk2-devel] [Patch v5 0/3] Add package level include DSC/FDF in NetworkPkg
http://mid.mail-archive.com/20190507142705.20092-1-liming.gao@intel.com
https://edk2.groups.io/g/devel/message/40112

Version 5 has been committed as:

1 da815cf390cc NetworkPkg DSC: Add the required ARM library to pass ARM build
2 1e948802e755 NetworkPkg: Add DSC/FDF include segment files to NetworkPkg.
3 23ea5bfea2a2 NetworkPkg: Add package level include DSC file

**Comment by tianocore-issues** [comment]
## Comment 7531
Date: 2019-05-14 10:41:43 +0000
From: @lersek

Platform patches dependent on the commits referenced in comment 6:

[edk2-devel] [PATCH 0/4] Add DSC/FDF include segment files for network stack
http://mid.mail-archive.com/20190514073938.21528-1-shenglei.zhang@intel.com
https://edk2.groups.io/g/devel/message/40583

**Comment by tianocore-issues** [comment]
## Comment 7553
Date: 2019-05-15 05:14:19 +0000
From: @lersek

Related patch:

[edk2-devel] [PATCH]
OvmfPkg: Remove duplicated library classes in dxe drivers section in DSC
http://mid.mail-archive.com/20190515072419.23792-1-shenglei.zhang@intel.com
https://edk2.groups.io/g/devel/message/40642

Commit a11d371ef660.

**Comment by tianocore-issues** [comment]
## Comment 7580
Date: 2019-05-16 10:01:04 +0000
From: @lersek

Platform patches v2:

[edk2-devel] [PATCH v2 0/4] Add Network&apos;s include segment files in DSC/FDF
http://mid.mail-archive.com/20190516081810.27840-1-shenglei.zhang@intel.com
https://edk2.groups.io/g/devel/message/40771

**Comment by tianocore-issues** [comment]
## Comment 7582
Date: 2019-05-16 11:04:12 +0000
From: @lersek

(In reply to Laszlo Ersek from comment #9)
&gt; Platform patches v2:
&gt; 
&gt; [edk2-devel] [PATCH v2 0/4] Add Network&apos;s include segment files in DSC/FDF
&gt; http://mid.mail-archive.com/20190516081810.27840-1-shenglei.zhang@intel.com
&gt; https://edk2.groups.io/g/devel/message/40771

Pushed as commit range 89d7c543cf71..3b7a897cd8e3.

**Comment by tianocore-issues** [comment]
## Comment 7730
Date: 2019-05-21 17:22:16 +0000
From: @lersek

As far as I can tell, this feature has missed edk2-stable201905 (some platform patches may still be missing, and the final code movement patches are very likely missing).

Please postpone the BZ reference from

  https://github.com/tianocore/tianocore.github.io/wiki/EDK-II-Release-Planning#edk2-stable201905-tag-planning

to

  https://github.com/tianocore/tianocore.github.io/wiki/EDK-II-Release-Planning#edk2-stable201908-tag-planning

Thanks,
Laszlo

**Comment by tianocore-issues** [comment]
## Comment 7817
Date: 2019-05-27 23:03:34 +0000
From: nobody &lt;<nobody>&gt;

Per comment https://edk2.groups.io/g/devel/message/41260?p=,,,20,0,0,0::Created,,Soft,20,2,0,31727161

This feature is committed to catch Q2 stable tag.

**Comment by tianocore-issues** [comment]
## Comment 8161
Date: 2019-06-24 11:29:12 +0000
From: @lersek

Final patches for this BZ:

* [edk2-devel] [Patch 0/3]
  Move network related components from MdeModulePkg to NetworkPkg
  http://mid.mail-archive.com/20190515120220.1112-1-liming.gao@intel.com
  https://edk2.groups.io/g/devel/message/40686

  Commit range c0fd7f734e2d..ec56fa278428.

* [edk2-devel] [Patch 0/3]
  Update Package DSC for Network module movement
  http://mid.mail-archive.com/20190528114204.16020-1-liming.gao@intel.com
  https://edk2.groups.io/g/devel/message/41498

  Commit range ec56fa278428..21d9dc21f818.
